### PR TITLE
Add: build and push helm chart action

### PR DESCRIPTION
**Pull request by pascalholthaus**
## What
Add: build and push helm chart action

## Why
DEVOPS-533

## References
DEVOPS-533

### CODE REVIEW

File: helm-build-push/action.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    default: ghcr.io
+    required: true
+  REGISTRY_URL:
+    description: Registry url
+    default: oci://ghcr.io/greenbone/helm-charts/
+    required: true
+  REGISTRY_USER:
+    description: Registry login user
+    required: true
+  REGISTRY_PASSWORD:
```
Comment (y0urself) [inline helm-build-push/action.yaml:22]:
```suggestion
  REGISTRY_TOKEN:
```

File: helm-build-push/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      - uses: actions/checkout@v3
+      - name: Build and push chart
+        uses: greenbone/actions/helm-build-push@v2
+        with:
+          CHART_NAME: Chart folder name
+          CHARTS_PATH: Charts base folder || default ./charts
+          REGISTRY: Registry to use || default ghcr.io
+          REGISTRY_URL: Registry url to push to e.g oci://ghcr.io/greenbone/helm-charts/
+          REGISTRY_USER: Registry username
+          REGISTRY_PASSWORD: Registry user password
```
Comment (y0urself) [inline helm-build-push/README.md:22]:
```suggestion
          REGISTRY_TOKEN: Registry user password
```

File: helm-build-push/action.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+runs:
+  using: "composite"
+  steps:
+    - name: Helm registry login 
+      shell: bash
+      run: |
+        helm registry login \
+          -u '${{ inputs.REGISTRY_USER }}' \
+          -p '${{ inputs.REGISTRY_PASSWORD }}' \
```
Comment (y0urself) [inline helm-build-push/action.yaml:37]:
```suggestion
          -p '${{ inputs.REGISTRY_TOKEN }}' \
```

**Review by y0urself — COMMENTED** [review]
Isn't it a TOKEN instead of a password?

File: helm-build-push/action.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  CHARTS_PATH:
+    description: Path to charts base folder
+    default: ./charts
+    required: false
+  CHART_NAME:
+    description: "Chart to build and push"
+    required: true
+  REGISTRY:
+    description: Registry
+    default: ghcr.io
```
Comment (y0urself) [inline helm-build-push/action.yaml:13]:
```suggestion
    default: ${{ vars.IMAGE_REGISTRY }}
```

File: helm-build-push/action.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    required: true
+  REGISTRY_URL:
+    description: Registry url
+    default: oci://ghcr.io/greenbone/helm-charts/
```
Comment (y0urself) [inline helm-build-push/action.yaml:17]:
```suggestion
    default: oci://${{ vars.IMAGE_REGISTRY }}/greenbone/helm-charts/
```

File: helm-build-push/action.yaml (hunk 4 above)
Comment (pascalholthaus) [reply helm-build-push/action.yaml:13]: You cannot use repository vars in actions

File: helm-build-push/action.yaml (hunk 5 above)
Comment (pascalholthaus) [reply helm-build-push/action.yaml:17]: You cannot use repository vars in actions

File: helm-build-push/action.yaml (hunk 1 above)
Comment (pascalholthaus) [reply helm-build-push/action.yaml]: It can be both.

File: helm-build-push/action.yaml (hunk 4 above)
Comment (y0urself) [reply helm-build-push/action.yaml:13]: I use them too, and this is a global variable.
[This](https://github.com/greenbone/actions/blob/main/lock-branch/action.yaml#L15) for example is working.

Did you test it?

**Review by y0urself — APPROVED** [review]

**Pull request merged**